docs/user-guide/petset.md

```diff
@@ -0,0 +1,269 @@
+---
+---
+
+* TOC
+{:toc}
+
+__Terminology and conventions__
+
+Throughout this doc you will see a few terms that are sometimes used interchangeably elsewhere, that might cause confusion. This section attempts to clarify them.
+
+* Node: A single virtual or physical machine in a Kubernetes cluster.
+* Cluster: A group of nodes in a single failure domain, unless mentioned otherwise.
+* Persistent Volume Claim: A request for storage, typically a [persistent volume](/docs/user-guide/persistent-volumes/walkthrough.md).
+* Host name: The hostname attached to the UTS namespace of the pod, i.e the output of `hostname` in the pod.
+* DNS/Domain name: A *cluster local* domain name resolvable using standard methods (eg: [gethostbyname](http://linux.die.net/man/3/gethostbyname)).
+* Ordinality: the proprety of being "ordinal", or occupying a position in a sequence.
+* Pet: a single member of a Pet Set; more generally, a stateful application.
+* Peer: a process running a server, capable of communicating with other such processes.
+
+__Prerequisites__
+
+This doc assumes familiarity with the following Kubernetes concepts:
+
+* [Pods](/docs/user-guide/gc/pods/single-container.md)
+* [Cluster DNS](/docs/admin/dns.md)
+* [Headless Services](/docs/user-guide/gc/services/#headless-services)
+* [Persistent Volumes](/docs/user-guide/gc/volumes.md)
+* [Dynamic volume provisioning](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md)
+
+You need a working Kubernetes cluster at version >= 1.3, with a healthy DNS [cluster addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/README.md) at version >= 15. You cannot use Pet Set on a hosted Kubernetes provider that has disabled `alpha` resources.
+
+## What is a Pet Set?
+
+In Kubernetes, most pod management abstractions group them into disposable units of work that compose a micro service. Replication controllers for example, are designed with a weak guarantee - that there should be N replicas of a particular pod template. The pods are treated as stateless units, if one of them is unhealthy or superseded by a better version, the system just disposes it.
```

Comment:
What is a pod management abstraction?  Is this a term we use elsewhere?  Are these Controllers?